packages/x-date-pickers/src/internals/utils/date-utils.ts

```diff
@@ -177,7 +177,7 @@ export const resolveDateFormat = <TDate extends PickerValidDate>(
   }
 
   if (areViewsEqual(views, ['day'])) {
-    return formats.dayOfMonth;
+    return formats.keyboardDate;
```

Comment:
We could perhaps remove this case and let it return the default format, do you see any problems with that ?